FW/Presentations/Lab_Guides/_L_C_04_Writing_UEFI_App_Simics_Linux_Lab.md

```diff
@@ -0,0 +1,2024 @@
+
+The build for BoardX58Ich10 OpenBoardPkg is using build MACRO Switch: `-D ADD_SHELL_STRING` - used to change a string in the UEFI Shell application, only used for EDK II Training (requirs ShellPkg be re-built on a change of this switch)
+
+
+---
+## Slide 34 Lab 2.1: Compiling w/ Build Switch
+
+Result without the build switch
+
+```
+Shell> ver
+UEFI Interactive Shell v2.2
+EDK II
+UEFI v2.70 (EDK II, 0x00010000)
+Shell>
+```
+
+Result with the build switch
+
+```
+Shell> ver
+UEFI Interactive Shell v2.2 -From ADD_SHELL_STRING Switch
+EDK II
+UEFI v2.70 (EDK II, 0x00010000)
+Shell> 
+```
+
+---
+## Slide 35 Lab 2.1: Compiling w/ Build Switch
+
+### Check Result without the build switch
+
+Run the qsp-modern-core script from Terminal Command Prompt:
+
+```bash
+$> ./simics targets/qsp-x86/qsp-modern-core.simics
+simics> run
+```
+
+Press "F2" at the logo, then select "Boot Manager" followed by "EFI Internal Shell"
+
+At the UEFI Shell prompt
+
+```
+Shell> ver
+UEFI Interactive Shell v2.2
+EDK II
+UEFI v.2.70 (EDK II, 0x00010000)
+Shell>
+```
+
+Exit Simics
+
+```
+simics> stop
+simics> quit
+```
+
+---
+## Slide 36 Lab 2.1: Compiling w/ Build Switch
+
+### Add the Build Switch `-D ADD_SHELL_STRING`
+
+**Two Ways:**
+
+1. Update Python Script or
+2. Add list of DEFINES in .DSC file (Preferred)
+
+
+### Method 1
+
+The Build command is part of the python script, build_bios.py
+
+Notice the Build command from the python script for BoardX58Ich10:
+
+```
+...
+Calling build -n 0 --log=Build.log --report-file=BuildReport.log
+...
+```
+
+Update line 388 of build_bios.py to include `-D ADD_SHELL_STRING`
+
+```python
+command = ["build", "-n", config["NUMBER_OF_PROCESSORS"], "-D", "ADD_SHELL_STRING"]
+```
+
+Re-Build BoardX58Ich10
+From a Terminal Command Prompt
+
+```bash
+$> cd ~/fw/edk2-ws/edk2-platforms/Platform/Intel/
+$> python build_bios.py -p BoardX58Ich10 GCC5
+```
+
+```
+...
+Calling build -n 0 -D ADD_SHELL_STRING --log=Build.log --report-file=BuildReport.log
+...
+```
+
+---
+## Slide 37 Lab 2.1: Compiling w/ Build Switch
+
+1. Invoke Simics and Test Shell Ver Command
+
+Copy `~/fw/edk2-ws/Build/SimicsOpenBoardPkg/BoardX58Ich10/DEBUG_GCC5/FV/BOARDX58ICH10.fd` To `Simics-Install-Directory/simics-qsp-x86-6.0.57/targets/qsp-x86/images`
+
+Run the qsp-modern-core script from Terminal Command Prompt
+
+```bash
+$> ./simics targets/qsp-x86/qsp-modern-core.simics
+simics> run
+``` 
+
+Press "F2" at the logo, then select "Boot Manager" followed by "EFI Internal Shell"
+
+At the UEFI Shell prompt, type: ver
+
+```
+Shell> ver
+UEFI Interactive Shell v2.2 -From ADD_SHELL_STRING Switch
+EDK II
+UEFI v2.70 (EDK II, 0x00010000)
+Shell>
+```
+
+Exit Simics
+
+```
+simics> stop
+simics> quit
+```
+
+---
+## Slide 38 Lab 2.1: Compiling w/ Build Switch
+
+### Method 2
+
+Add list of DEFINES in .DSC file. This is the **preferred** method for EDK II when a Build script is used
+
+Update line 338 of build_bios.py and remove the `-D ADD_SHELL_STRING` - Save 
```

Comment:
1. Update line 388 of  build_bios.py and remove the “-D ADD_SHELL_STRING” , then Save
2. Edit the file  `~/FW/edk2-ws/edk2-platforms/Platform/Intel/SimicsOpenBoardPkg/BoardX58Ich10/OpenBoardPkgBuildOption.dsc` 
* add the following DEFINE after Line 7
* Save after update